lib/text_cleaner.rb

```diff
@@ -27,9 +31,9 @@ def self.clean(text, opts = {})
     # Replace ????? with a single ?
     text.gsub!(/\?+/, '?') if opts[:deduplicate_question_marks]
     # Replace all-caps text with regular case letters
-    text.tr!('A-Z', 'a-z') if opts[:replace_all_upper_case] && (text =~ /[A-Z]+/) && (text == text.upcase)
+    text = text.mb_chars.downcase.to_s if opts[:replace_all_upper_case] && (text =~ /[A-Z]+/) && (text == text.upcase)
     # Capitalize first letter, but only when entire first word is lowercase
-    text.sub!(/\A([a-z]*)\b/) { |first| first.capitalize } if opts[:capitalize_first_letter]
+    text = text.mb_chars.capitalize.to_s if opts[:capitalize_first_letter] && (text =~ /\S+/) && text.split(' ').first == text.split(' ').first.mb_chars.downcase
```

Comment:
a bit confusing and I am pretty sure it is broken and capitalizing full sentence :) something like this? 

```
first, rest = text.split(' ',2)
if opts[:capitalize_first_letter] && first == first.mb_chars.downcase
   text = "#{first.mb_chars.capitalize}#{rest?" " + rest : ""}" 
end
```